### PR TITLE
validate-BA108

### PR DESCRIPTION
**Pull request by YairGlik**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9862

### CODE REVIEW

File: demisto_sdk/commands/validate/tests/test_tools.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def create_integration_object(
     values: Optional[List[Any]] = None,
     pack_info: Optional[Dict[str, Any]] = None,
     readme_content: Optional[str] = None,
+    integration_folder_name: Optional[str] = None,
```
Comment (jbabazadeh) [inline demisto_sdk/commands/validate/tests/test_tools.py:35]: Rename this to `name`

File: demisto_sdk/commands/validate/tests/test_tools.py [hunk 2]
```diff
@@ -182,6 +186,7 @@ def create_script_object(
     paths: Optional[List[str]] = None,
     values: Optional[List[Any]] = None,
     pack_info: Optional[Dict[str, Any]] = None,
+    script_folder_name: Optional[str] = None,
```
Comment (jbabazadeh) [inline demisto_sdk/commands/validate/tests/test_tools.py:189]: same here

File: demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        separators = ["_", "-"]
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(content_item.path.parent.name),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (any(char in separators for char in content_item.path.parent.name))
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py:34]:
```suggestion
            if (any(separator in content_item.path.parent.name for separator in separators))
```

File: demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+ContentTypes = Union[Integration, Script]
+
+
+class IsFolderNameHasSeparatorsValidator(BaseValidator[ContentTypes]):
+    error_code = "BA108"
+    description = "Check if there are separators in the folder name."
+    error_message = "The folder name '{0}' should be without any separator."
+    related_field = ""
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py:20]:
```suggestion
    related_field = "file path"
```

File: demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    rationale = "To ensure consistent, readable folder structures by avoiding separators like spaces, underscores, or hyphens."
+    is_auto_fixable = False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py:22]:
```suggestion
```

File: demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Union[Integration, Script]
+
+
+class IsFolderNameHasSeparatorsValidator(BaseValidator[ContentTypes]):
+    error_code = "BA108"
+    description = "Check if there are separators in the folder name."
+    error_message = "The folder name '{0}' should be without any separator."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA108_is_folder_name_has_separators.py:19]:
```suggestion
    error_message = "The folder name '{0}' should be without any separator."
```
Let's add an example of the separators we're talking about.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 7]
```diff
@@ -1867,3 +1870,43 @@ def test_IsContentItemNameContainTrailingSpacesValidator_fix(
...
+            ],
+            "",
+            id="a valid script name",
+        ),
+    ],
+)
+def test_IsFolderNameHasSeparatorsValidator_is_valid(content_items, expected_msg):
+    result = IsFolderNameHasSeparatorsValidator().is_valid(content_items)
+
+    if result:
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1911]:
```suggestion
    result = IsFolderNameHasSeparatorsValidator().is_valid(content_items)

    if result:
```
Better to test the number of results here as well
If results = [] in the first case then we'll still pass the test.

**Review by YuvHayun — APPROVED** [review]
Great job!
Minor comments.
@jbabazadeh approved from my side.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py (hunk 7 above)
Comment (YairGlik) [reply demisto_sdk/commands/validate/tests/BA_validators_test.py:1911]: I added,
I hope that's what you meant

**Pull request merged**